### DISCUSSION TITLE
Can we approximate measurements of 3D objects such as cubes or cube-like figures using the Kornia library?

### DISCUSSION THREAD

**Discussion by XinyueZ** [kornia/kornia#2647] [Questions]
Hey,

I possess two depth-cameras, mounted atop a door or gate, left and right. An object, resembling a cube or similar shape, passes through and is captured in RGBD by the cameras, so, 2 images.
Can we approximate measurements of 3D objects such as cubes or cube-like figures using the Kornia library?

```
cam1 ---------------------- cam2
|                              |
|                              |
|                              |


````

**Comment by edgarriba** [comment]
one possibility could be to compute a pointcloud from the depth maps using [`depth_to_3d`](https://kornia.readthedocs.io/en/latest/geometry.depth.html#kornia.geometry.depth.depth_to_3d) to have a more geometric representation of each of the views or even compute the normals via [`depth_to_normals`](https://kornia.readthedocs.io/en/latest/geometry.depth.html#kornia.geometry.depth.depth_to_normals) and based on that check whether you have planar surfaces (via ransac?) to isolate the planes. Another option could be using `ICP` to match a temple of your cube.

**We don't have ICP but could be very interesting to have /cc @ducha-aiki
